demos/src/Provisioning/aws_iot_demo_provisioning_with_csr.c

```diff
@@ -0,0 +1,594 @@
+                "StatusCode={%d}", pResponseInfo->statusCode );
+
+    if( pResponseInfo->statusCode == AWS_IOT_PROVISIONING_SERVER_STATUS_ACCEPTED )
+    {
+        /* Allocate buffer space for storing the certificate ID obtained from the server. */
+        certIdTokenContext->pCertificateIdBuffer =
+            Iot_DefaultMalloc( pResponseInfo->u.acceptedResponse.certIdLength + 1 );
+
+        /* Copy the certificate ID into the buffer. */
+        if( certIdTokenContext->pCertificateIdBuffer != NULL )
+        {
+            /* Copy the size of the Certificate ID string. */
+            certIdTokenContext->certificateIdLength = pResponseInfo->u.acceptedResponse.certIdLength;
+
+            memcpy( certIdTokenContext->pCertificateIdBuffer,
+                    pResponseInfo->u.acceptedResponse.pCertId,
+                    pResponseInfo->u.acceptedResponse.certIdLength );
+            /* Add a NULL terminator to the buffer (to treat the buffer as a string!) */
+            *( certIdTokenContext->pCertificateIdBuffer + pResponseInfo->u.acceptedResponse.certIdLength ) = '\0';
+        }
+
+        /* Allocate buffer space for storing the ownership token string obtained from the server. */
+        certIdTokenContext->pOwnershipTokenBuffer =
+            Iot_DefaultMalloc( pResponseInfo->u.acceptedResponse.ownershipTokenLength + 1 );
+
+        /* Copy the ownership token into the buffer. */
+        if( certIdTokenContext->pOwnershipTokenBuffer != NULL )
+        {
+            /* Copy the size of the ownership token string. */
+            certIdTokenContext->tokenLength = pResponseInfo->u.acceptedResponse.ownershipTokenLength;
+
+            memcpy( certIdTokenContext->pOwnershipTokenBuffer,
+                    pResponseInfo->u.acceptedResponse.pCertOwnershipToken,
+                    pResponseInfo->u.acceptedResponse.ownershipTokenLength );
+            /* Add a NULL terminator to the buffer (to treat the buffer as a string!) */
+            *( certIdTokenContext->pOwnershipTokenBuffer + pResponseInfo->u.acceptedResponse.ownershipTokenLength ) = '\0';
+        }
+
+        /* Print the received credentials information.*/
+        IotLogInfo( "demo: Server has accepted Certificate-Signing Request: "
+                    "\nCertificate PEM={%.*s}\nCertificate ID={%.*s}\nOwnership Token={%.*s}\n",
+                    pResponseInfo->u.acceptedResponse.deviceCertLength,
+                    pResponseInfo->u.acceptedResponse.pDeviceCert,
+                    pResponseInfo->u.acceptedResponse.certIdLength,
+                    pResponseInfo->u.acceptedResponse.pCertId,
+                    pResponseInfo->u.acceptedResponse.ownershipTokenLength,
+                    pResponseInfo->u.acceptedResponse.pCertOwnershipToken );
+    }
+    else
+    {
+        IotLogWarn( "demo: Oops, certificate creation from CSR has been rejected. "
+                    "Consider re-running the demo!" );
```

Comment:
"Consider re-running the demo!" --> Does not sound like a good message and I see that at multiple places.

Reply:
Have updated the log messages to be helpful by suggesting to verify the attached IoT Policy for the demo credentials